### ISSUE TITLE
Stylelint is linting CSS inside my comments

### ISSUE THREAD

**Issue by onlywei** [stylelint/stylelint#3062]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug? Feature? Not sure, but how do I turn it off?

> Which rule, if any, is this issue related to?

declaration-block-semicolon-newline-after: always
declaration-block-trailing-semicolon: always

> What CSS is needed to reproduce this issue?

```css
/*
Write this in a comment:
<div style="display: flex; flex-direction: row">
*/
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "declaration-block-semicolon-newline-after": "always",
    "declaration-block-trailing-semicolon": "always"
  }
}
```

> Which version of stylelint are you using?

`8.3.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Both with CLI, and in the demo at https://stylelint.io/demo/

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

I don't expect stylelint to be looking for CSS in my comments, especially in HTML `style` attributes.

> What actually happened (e.g. what warnings or errors you are getting)?

Stylelint is linting HTML style tags inside my comments.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Indeed it's very odd behavior. We need to investigate further why it happens.

**Comment by hudochenkov** [comment]
Hm. I can reproduce it in the demo, but can't reproduce with CLI using stylelint@8.3.1.

**Comment by onlywei** [comment]
Are you using a fresh install? Do you need to re-generate your `package-lock.json` or `yarn.lock` file?

**Comment by hudochenkov** [comment]
Yes, it's a fresh install. To be sure that stylelint works also added code with violations to test file and saw errors in my code only.

**Comment by hudochenkov** [comment]
I was able to reproduce a problem with Node API.

```js
const stylelint = require('stylelint');

const config = {
	rules: {
		'declaration-block-semicolon-newline-after': 'always',
		'declaration-block-trailing-semicolon': 'always',
	},
};

const code = `
/*
Write this in a comment:
<div style="display: flex; flex-direction: row">
*/
`;

stylelint
	.lint({
		formatter: 'string',
		code,
		config,
	})
	.then(
		function(data) {
			console.log(data.output);
		},
		function(err) {
			console.log(err);
		}
	);
```

```
$ node run-my.js

<input css 2>
 4:27  ✖  Expected newline after ";"      declaration-block-semicolon-newline-after
 4:46  ✖  Expected a trailing semicolon   declaration-block-trailing-semicolon
```

@onlywei how do you run stylelint? What extension file with a problem have?

**Comment by hudochenkov** [comment]
After some research, I found that files with `.css` and `.scss` extensions aren't affected. `.pcss` is affected.

Also, I found that this regression happened in 8.3.0. 8.2.0 doesn't have this problem.

All these details point to https://github.com/stylelint/stylelint/pull/2975. `postcss-html` doesn't recognize syntax correctly. @gucong3000 can you take a look, please?

@onlywei as a workaround until we fix a problem, could you try to use `--syntax scss`?

```
stylelint '**/*.*css' --syntax scss
```

**Comment by gucong3000** [comment]
This is a bug of `postcss-html`, I'll deal with it today. https://github.com/gucong3000/postcss-html/issues/18

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/3064

Will be included in `8.4.0`, which I intend to release now.
